targets/unity-v2/source/Shared/Public/PlayFabDataGatherer.cs

```diff
@@ -97,7 +97,11 @@ public void GatherData()
             DeviceName = SystemInfo.deviceName;
             DeviceType = SystemInfo.deviceType;
 
-            DeviceUniqueId = SystemInfo.deviceUniqueIdentifier;
+#if !DISABLE_UNIQUEIDENTIFIER
+            DeviceUniqueId = PlayFabSettings.DeviceUniqueIdentifier ?? SystemInfo.deviceUniqueIdentifier;
+#else
+            DeviceUniqueId = PlayFabSettings.DeviceUniqueIdentifier;
+#endif
```

Comment:
I'd rather have an ENABLE style #define, rather than a DISABLE, given the perceived risk of the permission it requires.
And I think the PlayFabSettings.DeviceUniqueIdentifier is clunky and deceptive.
Like many other fields in this file, it should simply not exist at all, if it's not available.